### PR TITLE
Feature/optimizations

### PR DESCRIPTION
**Pull request by rampr**
Minor optimizations, avoid loops, rename etc..

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -84,7 +84,9 @@ adv_map.AdventureMapLayer = cc.Layer.extend({
 
       map_data = cc.loader.getRes(this.data_path + tile + '.json');
       map_data.ms = ms;
-      cc.spriteFrameCache.addSpriteFrames(res[tile], res[tile + '_img']);
+      if(tile_sprites.indexOf(tile) === -1) {
```
Comment (hatim-developer) [inline index.js:87]: check if we are using underscore in adv map any where, then we can use _.constains, if indentation

File: index.js [hunk 2]
```diff
@@ -141,12 +146,6 @@ adv_map.AdventureMapLayer = cc.Layer.extend({
     return this.tile_layer_ref[adv_map.prefix.tile_layer + uniq_id];
   },
 
-  findTileLayerByMSNumber: function (ms) {
```
Comment (hatim-developer) [inline index.js:144]: did you checked we are not using this anywhere else right ? like in game

Comment (rampr) [reply index.js:144]: we are using in the game, have changed it to call `getTileLayerRef`
This function is just a wrapper to `getTileLayerRef`

Comment (hatim-developer) [reply index.js:144]: do u think this will help ?

File: index.js (hunk 1 above)
Comment (rampr) [reply index.js:87]: not using underscore in adventure map

File: index.js (hunk 2 above)
Comment (rampr) [reply index.js:144]: This is going to have barely any improvement in terms of performance. It makes the function generic.

**Review by hatim-developer — APPROVED** [review]

**Pull request merged**